### PR TITLE
fix: Slider show empty tooltip when tooltip.formatter is falsy

### LINKED ISSUE #48668: slider tooltip={{ formatter: null }} 隐藏 tooltip 后会有空的 tooltip 出现
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/p/sandbox/antd-reproduction-template-forked-zmsgzm?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clvgzryp000063b6lstaayht3%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clvgzryp000023b6lkt140p8v%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clvgzryp000033b6lbxbvj29w%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clvgzryp000053b6l4vm2ixsj%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clvgzryp000023b6lkt140p8v%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvgzryp000013b6lfm29wfhd%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A30%252C%2522startColumn%2522%253A23%252C%2522endLineNumber%2522%253A30%252C%2522endColumn%2522%253A23%257D%255D%257D%255D%252C%2522id%2522%253A%2522clvgzryp000023b6lkt140p8v%2522%252C%2522activeTabId%2522%253A%2522clvgzryp000013b6lfm29wfhd%2522%257D%252C%2522clvgzryp000053b6l4vm2ixsj%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvgzryp000043b6l2sloovlf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clvgzryp000053b6l4vm2ixsj%2522%252C%2522activeTabId%2522%253A%2522clvgzryp000043b6l2sloovlf%2522%257D%252C%2522clvgzryp000033b6lbxbvj29w%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clvgzryp000033b6lbxbvj29w%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Steps to reproduce

hover

### What is expected?

隐藏 tooltip

### What is actually happening?

有空的 tooltip

| Environment | Info |
| --- | --- |
| antd | 5.16.4 |
| React | 18.2.0 |
| System | macOS 版本14.4.1 (23E224) |
| Browser | chrome 版本 124.0.6367.80（正式版本） (arm64) |

---
文档中也可重现 https://ant.design/components/slider-cn#slider-demo-tip-formatter

image

### PR DESCRIPTION
**Pull request by wanpan11**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix https://github.com/ant-design/ant-design/issues/48668

### 💡 Background and solution

 formatter=null  时没有隐藏 tooltip

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix Slider tooltip cannot be hidden when `tooltip={{ formatter: null }}`.    |
| 🇨🇳 Chinese |  修复 Slider `tooltip={{ formatter: null }}` 无法隐藏 Tooltip 的问题。   |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/slider/__tests__/tooltip.test.tsx [hunk 1]
```diff
@@ -53,4 +53,25 @@ describe('Slider.Tooltip', () => {
     await waitFakeTimer();
     expect(tooltipProps().open).toBeFalsy();
   });
+
+  it('not show tooltip', async () => {
```
Comment (Wxh16144) [inline components/slider/__tests__/tooltip.test.tsx:57]:
```suggestion
  // https://github.com/ant-design/ant-design/issues/48668
  it('should not display tooltip when formatter is `null`', async () => {
```

Comment (Wxh16144) [reply components/slider/__tests__/tooltip.test.tsx:57]: 语义一下，或者贴个链接 方便追溯

Comment (wanpan11) [reply components/slider/__tests__/tooltip.test.tsx:57]: 好👌

File: components/slider/index.tsx [hunk 2]
```diff
@@ -291,14 +291,16 @@ const Slider = React.forwardRef<SliderRef, SliderSingleProps | SliderRangeProps>
 
     const cloneNode = React.cloneElement(node, passedProps);
 
+    const open = (!!lockOpen || activeOpen) && mergedTipFormatter !== null;
```
Comment (crazyair) [inline components/slider/index.tsx:294]: 要判断 null，还是假值，如果用户使用 undefined 是不是还会显示

Comment (wanpan11) [reply components/slider/index.tsx:294]: > 如果用户使用 undefined 是不是还会显示

是的 undefined 的话 就会显示内部的默认值

image

**Review by Wxh16144 — APPROVED** [review]
+1

**Review by afc163 — APPROVED** [review]

**Pull request merged**